pkg/deployment/autoscale.go

```diff
@@ -0,0 +1,107 @@
+package deployment
+
+import (
+	autoscalingv2beta2 "k8s.io/api/autoscaling/v2beta2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	"github.com/jaegertracing/jaeger-operator/pkg/util"
+)
+
+const (
+	// we need to have an upper bound, and 100 seems like a "good" max value
+	defaultMaxReplicas = int32(100)
+
+	// for both memory and cpu
+	defaultAvgUtilization = int32(90)
+)
+
+// Autoscalers returns a list of HPAs based on specs
+func autoscalers(
+	componentReplicas *int32,
+	label string,
+	componentName string,
+	componentLabels map[string]string,
+	autoScaleSpec v1.AutoScaleSpec,
+	componentSpec v1.JaegerCommonSpec,
```

Comment:
Would be better to call this `commonSpec` or `componentCommonSpec`?

Reply:
`componentCommonSpec` is a better name